src/main/java/com/example/bookstoreapp/mapper/BookMapper.java

```diff
@@ -2,14 +2,17 @@
 
 import com.example.bookstoreapp.config.MapperConfig;
 import com.example.bookstoreapp.dto.BookDto;
-import com.example.bookstoreapp.dto.CreateBookRequestDto;
+import com.example.bookstoreapp.dto.BookRequestDto;
 import com.example.bookstoreapp.model.Book;
 import org.mapstruct.Mapper;
+import org.mapstruct.MappingTarget;
 
 @Mapper(config = MapperConfig.class)
 public interface BookMapper {
 
     BookDto bookToBookDto(Book book);
 
-    Book createBookRequestDtoToBook(CreateBookRequestDto createBookRequestDto);
+    Book createBookRequestDtoToBook(BookRequestDto bookRequestDto);
+
+    void updateBookFromDto(BookDto bookDto, @MappingTarget Book book);
```

Comment:
```suggestion
    Book toBook(UpdateBookDto); //or toModel
```

Reply:
> 

I had some problems with it. With understanding, I could not find the way to do partial update with
Book toBook(UpdateBookDto);
AI suggest to select from mapper. But its is definitely bad suggestion. So I did two update methods. For patch and fotr put.